1_fetch/src/download_DO_data.R

```diff
@@ -0,0 +1,30 @@
+download_daily_mean_DO_data <- function(site_list,pcode,stat_cd,fileout){
+  
+  daily_DO_data_ls <- lapply(site_list,function(x){
+    readNWISdv(siteNumbers = x,parameterCd=pcode,statCd=stat_cd,startDate = "",endDate = "") %>%
+      renameNWISColumns()
+  })
+  
+  # Combine daily site data and save to fileout
+  daily_data_out <- do.call(rbind,daily_DO_data_ls)
+  
+  write_csv(daily_data_out, file = fileout)
+  
+  return(fileout)
+  
+}
+
+
+download_cont_DO_data <- function(site_list,pcode,time_zone){
+  
+  cont_DO_data_ls <- lapply(site_list,function(x){
```

Comment:
Can you just pass in a list of all the sites at once or do you have to do the `lapply` and then later `rbind`?

Reply:
Yep, you can pass in a list of all the sites at once and that would be more efficient here. I sometimes opt for the lapply-rbind steps if additional processing or variable name mungeing is needed on the downloaded data for individual sites, but that isn't happening here so I'll make the change to instead pass a list of sites to dataRetrieval. Thanks for pointing this out!

Reply:
One disadvantage of both the `lapply` way I had originally implemented and passing a list of all the sites to dataRetrieval is that _all_ of the data would be re-downloaded, for example, if just one site in the upstream site list changed. I learned some very useful tips and tricks from the last pipelines training course, so I'll use targets and dynamic branching for this sort of split-apply-combine workflow moving forward. I've gone ahead and implemented this sort of step with the daily data download and the `p1_daily_data` target. Now, if one site changes, we will only re-download the data for that one site rather than all sites. I'll use this same idea for the continuous data download, too.